src/main/java/org/openstreetmap/atlas/utilities/caching/LocalFileInMemoryCache.java

```diff
@@ -0,0 +1,77 @@
+package org.openstreetmap.atlas.utilities.caching;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.openstreetmap.atlas.streaming.resource.File;
+import org.openstreetmap.atlas.utilities.caching.strategies.ByteArrayCachingStrategy;
+import org.openstreetmap.atlas.utilities.caching.strategies.CachingStrategy;
+
+/**
+ * An example of how to extend the {@link ResourceCache} to enhance functionality. This class caches
+ * local files in memory, and abstracts the the messiness of file URIs behind a cleaner
+ * {@link LocalFileInMemoryCache#withPath(String)} interface. Note that it still must override the
+ * other "with" builder methods, so that the builder pattern does not throw up type errors. The
+ * {@link LocalFileInMemoryCache} forces the caching strategy to be
+ * {@link ByteArrayCachingStrategy}, and forces the default fetcher to simply load a local file.
+ *
+ * @author lcram
+ */
+public class LocalFileInMemoryCache extends ResourceCache
+{
+    /**
+     * Create a new {@link LocalFileInMemoryCache} with uninitialized default resource and fetcher.
+     */
+    public LocalFileInMemoryCache()
+    {
+        this(null);
+    }
+
+    /**
+     * Create a new {@link LocalFileInMemoryCache} with a given resource URI.
+     *
+     * @param resourceURI
+     *            the default resource {@link URI}
+     */
+    public LocalFileInMemoryCache(final URI resourceURI)
+    {
+        super(resourceURI, new ByteArrayCachingStrategy(), null);
```

Comment:
Any reason you are setting ```null```  ```defaultFetcher``` in the super call and then using the setter?

Reply:
I think this was just an oversight on my part. Good catch.